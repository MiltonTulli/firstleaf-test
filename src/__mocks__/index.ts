export const countries = [
  {
    name: 'Afghanistan',
    alpha2Code: 'AF',
    alpha3Code: 'AFG',
    capital: 'Kabul',
    region: 'Asia',
    subregion: 'Southern Asia',
    population: 27657145,
    latlng: [33, 65],
    demonym: 'Afghan',
    area: 652230,
    gini: 27.8,
    timezones: ['UTC+04:30'],
    borders: ['IRN', 'PAK', 'TKM', 'UZB', 'TJK', 'CHN'],
    nativeName: 'افغانستان',
    currencies: [
      {
        code: 'AFN',
        name: 'Afghan afghani',
        symbol: '؋'
      }
    ],
    languages: [
      {
        iso639_1: 'ps',
        iso639_2: 'pus',
        name: 'Pashto',
        nativeName: 'پښتو'
      },
      {
        iso639_1: 'uz',
        iso639_2: 'uzb',
        name: 'Uzbek',
        nativeName: 'Oʻzbek'
      },
      {
        iso639_1: 'tk',
        iso639_2: 'tuk',
        name: 'Turkmen',
        nativeName: 'Türkmen'
      }
    ],
    flag: 'https://restcountries.eu/data/afg.svg'
  },
  {
    name: 'Åland Islands',
    alpha2Code: 'AX',
    alpha3Code: 'ALA',
    capital: 'Mariehamn',
    region: 'Europe',
    subregion: 'Northern Europe',
    population: 28875,
    latlng: [60.116667, 19.9],
    demonym: 'Ålandish',
    area: 1580,
    gini: null,
    timezones: ['UTC+02:00'],
    borders: [],
    nativeName: 'Åland',
    currencies: [
      {
        code: 'EUR',
        name: 'Euro',
        symbol: '€'
      }
    ],
    languages: [
      {
        iso639_1: 'sv',
        iso639_2: 'swe',
        name: 'Swedish',
        nativeName: 'svenska'
      }
    ],
    flag: 'https://restcountries.eu/data/ala.svg'
  },
  {
    name: 'Albania',
    alpha2Code: 'AL',
    alpha3Code: 'ALB',
    capital: 'Tirana',
    region: 'Europe',
    subregion: 'Southern Europe',
    population: 2886026,
    latlng: [41, 20],
    demonym: 'Albanian',
    area: 28748,
    gini: 34.5,
    timezones: ['UTC+01:00'],
    borders: ['MNE', 'GRC', 'MKD', 'KOS'],
    nativeName: 'Shqipëria',
    currencies: [
      {
        code: 'ALL',
        name: 'Albanian lek',
        symbol: 'L'
      }
    ],
    languages: [
      {
        iso639_1: 'sq',
        iso639_2: 'sqi',
        name: 'Albanian',
        nativeName: 'Shqip'
      }
    ],
    flag: 'https://restcountries.eu/data/alb.svg'
  },
  {
    name: 'Algeria',
    alpha2Code: 'DZ',
    alpha3Code: 'DZA',
    capital: 'Algiers',
    region: 'Africa',
    subregion: 'Northern Africa',
    population: 40400000,
    latlng: [28, 3],
    demonym: 'Algerian',
    area: 2381741,
    gini: 35.3,
    timezones: ['UTC+01:00'],
    borders: ['TUN', 'LBY', 'NER', 'ESH', 'MRT', 'MLI', 'MAR'],
    nativeName: 'الجزائر',
    currencies: [
      {
        code: 'DZD',
        name: 'Algerian dinar',
        symbol: 'د.ج'
      }
    ],
    languages: [
      {
        iso639_1: 'ar',
        iso639_2: 'ara',
        name: 'Arabic',
        nativeName: 'العربية'
      }
    ],
    flag: 'https://restcountries.eu/data/dza.svg'
  }
];
