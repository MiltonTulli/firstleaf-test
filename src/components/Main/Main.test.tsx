import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { Main, CountryCard } from '..';
import { countries } from '../../__mocks__';

describe('<Main>', () => {
  let props;
  beforeEach(() => {
    props = {
      loading: false,
      error: false,
      countries: countries,
      handleSearch: jest.fn()
    };
  });
  it('Should match snapshot', () => {
    const wrapper = renderer.create(<Main {...props} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
  it('Should render cards', () => {
    const wrapper = shallow(<Main {...props} />);
    const node = wrapper.find(CountryCard);
    expect(node.length).toEqual(props.countries.length);
  });
});
