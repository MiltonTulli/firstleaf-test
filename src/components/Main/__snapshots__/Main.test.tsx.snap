// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main> Should match snapshot 1`] = `
<div
  className="makeStyles-root-1"
>
   
  <div
    className="makeStyles-root-6"
    data-testid="card"
    onClick={[Function]}
  >
    <div
      className="MuiPaper-root MuiCard-root makeStyles-root-6 MuiPaper-elevation10 MuiPaper-rounded"
    >
      <button
        className="MuiButtonBase-root MuiCardActionArea-root"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <div
          className="MuiCardContent-root"
        >
          <div
            className="makeStyles-contentWrapper-7"
          >
            <img
              alt="Afghanistan flag"
              className="makeStyles-flag-8"
              src="https://restcountries.eu/data/afg.svg"
            />
            <h1
              className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiTypography-alignCenter"
            >
              Afghanistan
            </h1>
          </div>
        </div>
        <span
          className="MuiCardActionArea-focusHighlight"
        />
      </button>
    </div>
  </div>
  <div
    className="makeStyles-root-6"
    data-testid="card"
    onClick={[Function]}
  >
    <div
      className="MuiPaper-root MuiCard-root makeStyles-root-6 MuiPaper-elevation10 MuiPaper-rounded"
    >
      <button
        className="MuiButtonBase-root MuiCardActionArea-root"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <div
          className="MuiCardContent-root"
        >
          <div
            className="makeStyles-contentWrapper-7"
          >
            <img
              alt="Åland Islands flag"
              className="makeStyles-flag-8"
              src="https://restcountries.eu/data/ala.svg"
            />
            <h1
              className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiTypography-alignCenter"
            >
              Åland Islands
            </h1>
          </div>
        </div>
        <span
          className="MuiCardActionArea-focusHighlight"
        />
      </button>
    </div>
  </div>
  <div
    className="makeStyles-root-6"
    data-testid="card"
    onClick={[Function]}
  >
    <div
      className="MuiPaper-root MuiCard-root makeStyles-root-6 MuiPaper-elevation10 MuiPaper-rounded"
    >
      <button
        className="MuiButtonBase-root MuiCardActionArea-root"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <div
          className="MuiCardContent-root"
        >
          <div
            className="makeStyles-contentWrapper-7"
          >
            <img
              alt="Albania flag"
              className="makeStyles-flag-8"
              src="https://restcountries.eu/data/alb.svg"
            />
            <h1
              className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiTypography-alignCenter"
            >
              Albania
            </h1>
          </div>
        </div>
        <span
          className="MuiCardActionArea-focusHighlight"
        />
      </button>
    </div>
  </div>
  <div
    className="makeStyles-root-6"
    data-testid="card"
    onClick={[Function]}
  >
    <div
      className="MuiPaper-root MuiCard-root makeStyles-root-6 MuiPaper-elevation10 MuiPaper-rounded"
    >
      <button
        className="MuiButtonBase-root MuiCardActionArea-root"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <div
          className="MuiCardContent-root"
        >
          <div
            className="makeStyles-contentWrapper-7"
          >
            <img
              alt="Algeria flag"
              className="makeStyles-flag-8"
              src="https://restcountries.eu/data/dza.svg"
            />
            <h1
              className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiTypography-alignCenter"
            >
              Algeria
            </h1>
          </div>
        </div>
        <span
          className="MuiCardActionArea-focusHighlight"
        />
      </button>
    </div>
  </div>
   
</div>
`;
